### ISSUE TITLE
xsession module doesn't handle parameter passed from display-manager such as sddm.

### ISSUE THREAD

**Issue by Vonfry** [nix-community/home-manager#2122]
### Issue description
display-manager includes two main parts when user login. One is session command, and the other is a desktop file containing `Exec` named `<de>+<wm>.desktop`, like `none+xmonad.desktop`, to handle different usage.

However, `de+wm.desktop` will do more rather than startup a WM or DE like nixos/nixpkgs#127367, wihch should be handled in hm. In addation, `de+wm.desktop` from nixos has some conflictions with `xsession` from hm.

In nixos module, a xsession is wrapped as follows:
```
...
if test -x ~/.xsession; then
    eval exec ~/.xsession "$@"
fi

if test "$1"; then
    # Run the supplied session command. Remove any double quotes with eval.
    eval exec "$@"
else
    # TODO: Do we need this? Should not the session always exist?
    echo "error: unknown session $1" 1>&2
    exit 1
fi
```
Apparently,  only either of them can be run.

In addition, the `~/.xsession` generated by hm does something which is also done in `de+wm.desktop` as follows:
```
${wm/de startup script}

/run/current-system/systemd/bin/systemctl --user stop graphical-session.target
```

I want to make hm handle the commands run in `de+wm.desktop`  and hm can config `xprofile` and `xsession` freely, but how? It is not a good idea to code every thing from nixos into hm.

There are several methods.

#### Seperate xsession in hm
Currently, hm codes `xprofile` and `xsession` with wm and user unit management togather. We can make things like following:
```
        ${cfg.initExtra}


if test "$1" then
       eval exec  "$@"
else
        ${cfg.windowManager.command}

        systemctl --user stop graphical-session.target
        systemctl --user stop graphical-session-pre.target

        # Wait until the units actually stop.
        while [ -n "$(systemctl --user --no-legend --state=deactivating list-units)" ]; do
          sleep 0.5
        done
fi
```
If user set de with nixos and they want to use wm with de by hm. They have to set wm command in `initExtra` manually.

#### run session command
```
       ...
        ${cfg.initExtra}
 
        waitPID=""
        if test  "${cfg.windowManager.command}" then
          ${cfg.windowManager.command} &
          waitPID=$!
        fi

        if test "$1"; then
            eval "$@"
        fi
        test -n "$waitPID" && wait "$waitPID"

        systemctl --user stop graphical-session.target
        systemctl --user stop graphical-session-pre.target

       ...
```
However, if user set same wm in both hm and nixos, the wm.command will conflict with the other. For example, they want to use hm to manage xmonad binary and nixos with xmonad as well. A solution may move the wm.command down or let user to ovrride manually with a document.
And we cannot test whether the parameter is passed as a command by dm or not. Some user may do things like `eval exec ~/.xsession foo bar` in `/etc/xsession` instead of the normal form `eval exec ~/.xsession "$@" `.

#### Only seperate xprofile from xsession module
Make xprofile working without current hm.xession module so that xsession in nixos can load variables set by hm. And the other challenges pushes to user. 
Currently, xprofile by hm must be generated with `xsession.enbale` which generates `xsession`.

#### Add default value to wm.command
Set the default value of `windowManager.command` to handle dm parameter. For user on nixos, he sets wm and de by nixos and xsession is a assistant with extra scripts. If he want to set use wm with hm instead of nixos module, let him handle the parameter by himself in `initExtra`.

I think it is the best method for both nixos and non-nixos.

**Comment by Vonfry** [comment]
I would like to make a pr, but I don't know which solution is better.

**Comment by berbiche** [comment]
> Add default value to wm.command

I prefer this solution too and I believe the one you implemented in #2123.
